packages/@sanity/default-layout/src/navbar/search/SearchField.tsx

```diff
@@ -1,111 +1,188 @@
-                <Hotkeys keys={hotkeys} />
-              </div>
-            )}
-            <div
-              className={value ? styles.clearButtonWithValue : styles.clearButton}
-              onClick={onClear}
-              title="Clear search"
+      if (!props.hidden && searchString && !loading && hits.length === 0) {
+        return (
+          <SearchFullscreenContent>
+            <Flex
+              align="center"
+              flex={1}
+              height="fill"
+              justify="center"
+              padding={4}
+              sizing="border"
             >
-              <CloseIcon />
-            </div>
-          </Layer>
-          <div className={styles.results}>{results}</div>
-        </div>
-      </LegacyLayerProvider>
-    )
+              <StyledText align="center" muted>
+                No results for <strong>‘{searchString}’</strong>
+              </StyledText>
+            </Flex>
+          </SearchFullscreenContent>
+        )
+      }
+
+      return (
+        <SearchFullscreenContent hidden={props.hidden} ref={ref}>
+          {props.content}
+        </SearchFullscreenContent>
+      )
+    },
+    [error, searchString, loading, hits]
+  )
+
+  const renderPopover = useCallback(
+    (props, ref) => {
+      if (!props.hidden && error) {
+        return (
+          <SearchPopoverContent
+            open={!props.hidden}
+            referenceElement={props.inputElement}
+            ref={ref}
+            content={
+              <Box padding={4}>
+                <Flex align="center" height="fill" justify="center">
+                  <StyledText align="center" muted>
+                    {error?.message || 'Something went wrong while searching'}
+                  </StyledText>
+                </Flex>
+              </Box>
+            }
+          />
+        )
+      }
+
+      if (!props.hidden && searchString && !loading && hits.length === 0) {
+        return (
+          <SearchPopoverContent
+            open={!props.hidden}
+            referenceElement={props.inputElement}
+            ref={ref}
+            content={
+              <Box padding={4}>
+                <Flex align="center" height="fill" justify="center">
+                  <StyledText align="center" muted>
+                    No results for <strong>‘{searchString}’</strong>
+                  </StyledText>
+                </Flex>
+              </Box>
+            }
+          />
+        )
+      }
+
+      if (!props.hidden && hits.length > 0) {
+        return (
+          <SearchPopoverContent
+            open={!props.hidden}
+            referenceElement={props.inputElement}
+            ref={ref}
+            content={props.content}
+          />
+        )
+      }
+
+      return undefined
+    },
+    [error, searchString, loading, hits]
+  )
+
+  const autoComplete = useMemo(
+    () => (
+      <Autocomplete
+        icon={SearchIcon}
+        key="studio-search"
+        id="studio-search"
```

Comment:
The search results listbox should have padding `2` on narrow screens.

```suggestion
        id="studio-search"
        listBox={{padding: 2}}
```